### ISSUE TITLE
libsymspg library loading error

### ISSUE THREAD

**Issue by fangohr** [fangohr/octopus-in-docker#7]
The [CI fails](https://github.com/fangohr/octopus-in-docker/actions/runs/7267472699/job/19801378669?pr=6#step:3:8557) with this error:

```
error while loading shared libraries: libsymspg.so.0: cannot open shared object file: No such file or directory\
```

This comes to light with the recent pull requests #5 and #6 but is more likely to be due to a change in the development version of Octopus.

**Comment by fangohr** [comment]
More details of the error (output from running `make develop` in https://github.com/fangohr/octopus-in-docker/pull/8 / branch `fix-libsymspg-error`.

```console
#17 0.205 DEBUG from here----------------XXXX
#17 DONE 0.2s

#18 [15/19] RUN ldd /usr/local/bin/octopus
#18 sha256:654864217432f5e49b5086702d5893b2e346bcd2208e1bab36caa5fdaf571e11
#18 0.140 	linux-vdso.so.1 (0x0000ffff88000000)
#18 0.140 	libsymspg.so.0 => not found
#18 0.140 	libnetcdff.so.7 => /usr/lib/aarch64-linux-gnu/libnetcdff.so.7 (0x0000ffff86d40000)
#18 0.140 	libnlopt.so.0 => /usr/lib/aarch64-linux-gnu/libnlopt.so.0 (0x0000ffff86cd3000)
#18 0.140 	libfftw3_omp.so.3 => /usr/lib/aarch64-linux-gnu/libfftw3_omp.so.3 (0x0000ffff86cbc000)
#18 0.140 	libfftw3.so.3 => /usr/lib/aarch64-linux-gnu/libfftw3.so.3 (0x0000ffff86bf1000)
#18 0.140 	liblapack.so.3 => /usr/lib/aarch64-linux-gnu/liblapack.so.3 (0x0000ffff86681000)
#18 0.140 	libblas.so.3 => /usr/lib/aarch64-linux-gnu/libblas.so.3 (0x0000ffff86637000)
#18 0.140 	libgsl.so.25 => /usr/lib/aarch64-linux-gnu/libgsl.so.25 (0x0000ffff8638f000)
#18 0.140 	libxcf03.so.5 => /usr/lib/aarch64-linux-gnu/libxcf03.so.5 (0x0000ffff86378000)
#18 0.140 	libstdc++.so.6 => /usr/lib/aarch64-linux-gnu/libstdc++.so.6 (0x0000ffff861a0000)
#18 0.140 	libgfortran.so.5 => /usr/lib/aarch64-linux-gnu/libgfortran.so.5 (0x0000ffff86036000)
#18 0.140 	libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000ffff85f8b000)
#18 0.140 	libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x0000ffff85f67000)
#18 0.140 	libmpi_mpifh.so.40 => /usr/lib/aarch64-linux-gnu/libmpi_mpifh.so.40 (0x0000ffff85ef2000)
#18 0.140 	libgomp.so.1 => /usr/lib/aarch64-linux-gnu/libgomp.so.1 (0x0000ffff85ea5000)
#18 0.140 	libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000ffff85e74000)
#18 0.140 	libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000ffff85d00000)
#18 0.140 	/lib/ld-linux-aarch64.so.1 (0x0000ffff87fd0000)
#18 0.140 	libnetcdf.so.18 => /usr/lib/aarch64-linux-gnu/libnetcdf.so.18 (0x0000ffff85bc1000)
#18 0.140 	libatlas.so.3 => /usr/lib/aarch64-linux-gnu/libatlas.so.3 (0x0000ffff85938000)
#18 0.140 	libgslcblas.so.0 => /usr/lib/aarch64-linux-gnu/libgslcblas.so.0 (0x0000ffff858f7000)
#18 0.140 	libxc.so.5 => /usr/lib/aarch64-linux-gnu/libxc.so.5 (0x0000ffff854e0000)
#18 0.140 	libmpi.so.40 => /usr/lib/aarch64-linux-gnu/libmpi.so.40 (0x0000ffff853ac000)
#18 0.140 	libopen-pal.so.40 => /usr/lib/aarch64-linux-gnu/libopen-pal.so.40 (0x0000ffff852e9000)
#18 0.140 	libdl.so.2 => /lib/aarch64-linux-gnu/libdl.so.2 (0x0000ffff852d5000)
#18 0.140 	libhdf5_serial_hl.so.100 => /usr/lib/aarch64-linux-gnu/libhdf5_serial_hl.so.100 (0x0000ffff852a3000)
#18 0.140 	libhdf5_serial.so.103 => /usr/lib/aarch64-linux-gnu/libhdf5_serial.so.103 (0x0000ffff84edf000)
#18 0.140 	libcurl-gnutls.so.4 => /usr/lib/aarch64-linux-gnu/libcurl-gnutls.so.4 (0x0000ffff84e3e000)
#18 0.140 	libopen-rte.so.40 => /usr/lib/aarch64-linux-gnu/libopen-rte.so.40 (0x0000ffff84d6c000)
#18 0.140 	libhwloc.so.15 => /usr/lib/aarch64-linux-gnu/libhwloc.so.15 (0x0000ffff84d0a000)
#18 0.140 	libutil.so.1 => /lib/aarch64-linux-gnu/libutil.so.1 (0x0000ffff84cf6000)
#18 0.140 	libevent_core-2.1.so.7 => /usr/lib/aarch64-linux-gnu/libevent_core-2.1.so.7 (0x0000ffff84cad000)
#18 0.140 	libevent_pthreads-2.1.so.7 => /usr/lib/aarch64-linux-gnu/libevent_pthreads-2.1.so.7 (0x0000ffff84c98000)
#18 0.140 	libsz.so.2 => /usr/lib/aarch64-linux-gnu/libsz.so.2 (0x0000ffff84c85000)
#18 0.140 	libz.so.1 => /lib/aarch64-linux-gnu/libz.so.1 (0x0000ffff84c5b000)
#18 0.140 	libnghttp2.so.14 => /usr/lib/aarch64-linux-gnu/libnghttp2.so.14 (0x0000ffff84c1f000)
#18 0.140 	libidn2.so.0 => /usr/lib/aarch64-linux-gnu/libidn2.so.0 (0x0000ffff84bef000)
#18 0.140 	librtmp.so.1 => /usr/lib/aarch64-linux-gnu/librtmp.so.1 (0x0000ffff84bc0000)
#18 0.140 	libssh2.so.1 => /usr/lib/aarch64-linux-gnu/libssh2.so.1 (0x0000ffff84b7e000)
#18 0.140 	libpsl.so.5 => /usr/lib/aarch64-linux-gnu/libpsl.so.5 (0x0000ffff84b5c000)
#18 0.140 	libnettle.so.8 => /usr/lib/aarch64-linux-gnu/libnettle.so.8 (0x0000ffff84b09000)
#18 0.140 	libgnutls.so.30 => /usr/lib/aarch64-linux-gnu/libgnutls.so.30 (0x0000ffff848ed000)
#18 0.140 	libgssapi_krb5.so.2 => /usr/lib/aarch64-linux-gnu/libgssapi_krb5.so.2 (0x0000ffff8488f000)
#18 0.140 	libldap_r-2.4.so.2 => /usr/lib/aarch64-linux-gnu/libldap_r-2.4.so.2 (0x0000ffff84828000)
#18 0.140 	liblber-2.4.so.2 => /usr/lib/aarch64-linux-gnu/liblber-2.4.so.2 (0x0000ffff84809000)
#18 0.140 	libbrotlidec.so.1 => /usr/lib/aarch64-linux-gnu/libbrotlidec.so.1 (0x0000ffff847ee000)
#18 0.140 	libudev.so.1 => /usr/lib/aarch64-linux-gnu/libudev.so.1 (0x0000ffff847b7000)
#18 0.140 	libaec.so.0 => /usr/lib/aarch64-linux-gnu/libaec.so.0 (0x0000ffff8479f000)
#18 0.140 	libunistring.so.2 => /usr/lib/aarch64-linux-gnu/libunistring.so.2 (0x0000ffff84611000)
#18 0.140 	libhogweed.so.6 => /usr/lib/aarch64-linux-gnu/libhogweed.so.6 (0x0000ffff845b9000)
#18 0.140 	libgmp.so.10 => /usr/lib/aarch64-linux-gnu/libgmp.so.10 (0x0000ffff84531000)
#18 0.140 	libgcrypt.so.20 => /usr/lib/aarch64-linux-gnu/libgcrypt.so.20 (0x0000ffff84461000)
#18 0.140 	libp11-kit.so.0 => /usr/lib/aarch64-linux-gnu/libp11-kit.so.0 (0x0000ffff84312000)
#18 0.140 	libtasn1.so.6 => /usr/lib/aarch64-linux-gnu/libtasn1.so.6 (0x0000ffff842ec000)
#18 0.140 	libkrb5.so.3 => /usr/lib/aarch64-linux-gnu/libkrb5.so.3 (0x0000ffff84203000)
#18 0.140 	libk5crypto.so.3 => /usr/lib/aarch64-linux-gnu/libk5crypto.so.3 (0x0000ffff841c5000)
#18 0.140 	libcom_err.so.2 => /lib/aarch64-linux-gnu/libcom_err.so.2 (0x0000ffff841b1000)
#18 0.140 	libkrb5support.so.0 => /usr/lib/aarch64-linux-gnu/libkrb5support.so.0 (0x0000ffff84194000)
#18 0.140 	libresolv.so.2 => /lib/aarch64-linux-gnu/libresolv.so.2 (0x0000ffff8416b000)
#18 0.140 	libsasl2.so.2 => /usr/lib/aarch64-linux-gnu/libsasl2.so.2 (0x0000ffff8413f000)
#18 0.140 	libbrotlicommon.so.1 => /usr/lib/aarch64-linux-gnu/libbrotlicommon.so.1 (0x0000ffff8410e000)
#18 0.140 	libgpg-error.so.0 => /lib/aarch64-linux-gnu/libgpg-error.so.0 (0x0000ffff840db000)
#18 0.140 	libffi.so.7 => /usr/lib/aarch64-linux-gnu/libffi.so.7 (0x0000ffff840c2000)
#18 0.140 	libkeyutils.so.1 => /lib/aarch64-linux-gnu/libkeyutils.so.1 (0x0000ffff840ab000)
#18 DONE 0.1s

#19 [16/19] RUN ldd /usr/local/bin/octopus | grep libsym
#19 sha256:58e3dec93cb5c5b4b6f37ec1717825930bb47d946f7b2c95fbbb35cf50598742
#19 0.289 	libsymspg.so.0 => not found
#19 DONE 0.3s

#20 [17/19] RUN dpkg -L libsymspg1
#20 sha256:ed06e2d6267d69571209b53e62e2d2b4430093d5dbee6a15ef1ce79f7d444c85
#20 0.327 /.
#20 0.327 /usr
#20 0.327 /usr/lib
#20 0.327 /usr/lib/aarch64-linux-gnu
#20 0.327 /usr/lib/aarch64-linux-gnu/libsymspg.so.1.16.1
#20 0.327 /usr/share
#20 0.327 /usr/share/doc
#20 0.327 /usr/share/doc/libsymspg1
#20 0.327 /usr/share/doc/libsymspg1/changelog.Debian.gz
#20 0.327 /usr/share/doc/libsymspg1/changelog.gz
#20 0.327 /usr/share/doc/libsymspg1/copyright
#20 0.327 /usr/lib/aarch64-linux-gnu/libsymspg.so.1
#20 DONE 0.3s

#21 [18/19] RUN dpkg -L libsymspg-dev
#21 sha256:c50a8320f4017ee509ec66c7c3b8ba26ac11add7f6c2ca4a2774952c889ac8c0
#21 0.140 /.
#21 0.140 /usr
#21 0.140 /usr/include
#21 0.140 /usr/include/spglib.h
#21 0.140 /usr/lib
#21 0.140 /usr/lib/aarch64-linux-gnu
#21 0.140 /usr/lib/aarch64-linux-gnu/libsymspg.a
#21 0.140 /usr/share
#21 0.140 /usr/share/doc
#21 0.140 /usr/share/doc/libsymspg-dev
#21 0.140 /usr/share/doc/libsymspg-dev/changelog.Debian.gz
#21 0.140 /usr/share/doc/libsymspg-dev/changelog.gz
#21 0.140 /usr/share/doc/libsymspg-dev/copyright
#21 0.140 /usr/lib/aarch64-linux-gnu/libsymspg.so
#21 DONE 0.1s

```

**Comment by fangohr** [comment]
It looks like Octopus is looking for `libsymspg.so.0` but only `libsymspg.so.1.*` is available.

**Comment by fangohr** [comment]
The failure occurs for the development version of octopus. The latest release compiles fine (see https://github.com/fangohr/octopus-in-docker/actions/runs/7268289955/job/19803823120#step:3:13282 ).

**Comment by fangohr** [comment]
I close this ticket. I cleaner more distinct summary is available in #9 .
